public_html/wp-content/themes/pattern-directory/src/components/patterns/index.js

```diff
@@ -19,6 +21,7 @@ const Patterns = () => {
 	return (
 		<RouteProvider>
 			<QueryMonitor />
+			<BreadcrumbMonitor />
 			<PatternGridMenu />
```

Comment:
I still think you could get rid of `PatternsSearch`, you'd just need to check for a search value before showing the menu. What do you think?

```js
const query = useSelect( ( select ) => select( patternStore ).getCurrentQuery() );
// Only show the full menu if the query does not have a search term.
const showMenu = ! query?.search;

return (
	<RouteProvider>
		<QueryMonitor />
		<BreadcrumbMonitor />
		{ showMenu ? <PatternGridMenu /> : <CategoryContextBar /> }
```

Reply:
Right, it was a bit of a toss up but I chose a separate component because of the yet to be implemented search error page:

![](https://d.pr/i/YRUNxj.png)

We could pass an empty results views to the pattern grid but historically I have run into more problems by passing components than I have composing components.

Reply:
I have another PR that I just need to write up for the empty results here, but it's the same empty grid as when there are no items in a category, so I still don't think this needs a separate component -- but I don't think I follow your second comment.

Reply:
When viewing `author/wordpressdotorg/`, clicking on any of the categories changes the URL from `author/wordpressdotorg/` to `pattern-categories/{category}/`. 

We have an issue to address that: [issue](268).

We'll need to figure out a new URL structure.  Maybe: `author/wordpressdotorg/{category}`. 

The `PatternGridMenu`, will probably not work with that new URL structure by default. It will need to be refactored if we want to use it. Refactoring may lead to conditional views, which I don't like too much. I think composing a new `PatternGridMenu` maybe `AuthorPatternGridMenu` to keep the controls simple will be a better approach. 

If we went with that approach, having an `AuthorPatterns` will allow us to compose that page, and then we don't have to include the `BreadcrumbMonitor` because `Author: Wordpress.org` or `Author: Wordpress.org : Gallery` is not a great breadcrumb. It would probably be something like `Author / Wordpress.org / Gallery`.....

.... hope I'm explaining my thoughts well enough.. regardless, I don't have any changes on the PR that I'm waiting on.

Reply:
I think we just disagree in approach. I would rather see `PatternGridMenu` used on the author page too, to create URLs like `/author/wordpressdotorg/gallery`, or even `/author/wordpressdotorg/?pattern-category=gallery` if the rewrite is tricky.

On the breadcrumb, I think it should be `All Patterns / Author: Wordpress.org / Gallery` - but that can be done in `BreadcrumbMonitor`, we don't need to create a special `AuthorBreadcrumbMonitor`.